components/reddit/sources/new-link-on-a-subreddit/new-link-on-a-subreddit.js

```diff
@@ -0,0 +1,84 @@
+const reddit = require("../../reddit.app.js");
+const get = require("lodash.get");
+
+module.exports = {
+  key: "new-link-on-a-subreddit",
+  name: "New Link on a subreddit",
+  description: "Emmits a new link submitted to a subreddit",
+  version: "0.0.2",
+  props: {
+    reddit,
+    timer: {
+      label: "Polling schedule",
+      description:
+        "Pipedream poll for new links and emit accordingly with this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 10, // by default, run every 10 minute
+      },
+    },
+    subreddit: {
+      type: "string",
+      label: "Subreddit",
+      description: "Enter subreddit name to listen for new links.",
+      default: "redditdev",
+    },
+    db: "$.service.db",
+  },
+  hooks: {
+    async deploy() {
+      // Emmits all existing events for the first time.
+      var after = null;
+      do {
+        const reddit_things = await this.reddit.getNewSubredditLinks(
+          null,
+          this.subreddit
+        );
+        if (get(reddit_things, "data.after") !== undefined) {
+          after = reddit_things.data.after;
+          console.log(
+            `after children length: ${reddit_things.data.children.length}`
+          );
+          if (reddit_things.data.children.length > 0) {
+            reddit_things.data.children.forEach((reddit_link) => {
+              this.emitMeInfo(reddit_link);
+            });
+          }
+        }
+      } while (after);
+      this.db.set("after", after);
+      console.log(`finished [deploy] hook, after: ${after}`);
+    },
+  },
+  methods: {
+    emitMeInfo(reddit_link) {
+      this.$emit(reddit_link, {
+        summary: reddit_link.data.name,
```

Comment:
Did you happen to see the docs on the `id` property of the emit metadata? If Reddit assigns a unique ID to links, you can pass that here. Then, you can use the `dedupe: "unique"` strategy, and Pipedream won't emit events with the same ID twice. Check out https://github.com/PipedreamHQ/pipedream/blob/master/COMPONENT-API.md#dedupe-strategies .

Reply:
Yes, I saw them. As to what I understood on the reddit api docs, reddit is supposed to guarantee unique ids among calls provided a greater "after" is provided in each call. I added the dedupe strategy which is a good fit anyway since after is contains a link full name, which is global id among all reddit object types.